examples/controller_tests/testdata/go.mod

```diff
@@ -1,77 +1,82 @@
 module tutorial.kubebuilder.io/project
 
-go 1.20
+go 1.22.3
```

Comment:
bump to 1.23

Reply:
Do we really need this, go version 1.22.3 came from the e2e-framework v0.5.0 in go.mod automatically and aligned with all components? Is there any plans to bump golang verison in project go.mod soon?

Reply:
i can do in a follow up

Reply:
Ok, fixed